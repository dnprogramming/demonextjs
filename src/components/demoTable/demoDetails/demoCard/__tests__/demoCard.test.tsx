import { render, screen } from "@testing-library/react";
import DemoCard from "../demoCard";
import React from "react";
import DemoContent from "@/content/demoTableData";

describe("DemoCard Component", () => {
    it("renders DemoCard description", () => {
        render(<DemoCard demoData={DemoContent[0].demoData} />);
        const headerElement = screen.getByText(DemoContent[0].demoData[0].description);
        expect(headerElement).toBeInTheDocument();
    });


    it("renders DemoCard list items", () => {
        render(<DemoCard demoData={DemoContent[0].demoData} />);
        DemoContent[0].demoData.forEach((demoInfo) => {
                const listItemElement = screen.getByText(demoInfo.description);
                expect(listItemElement).toBeInTheDocument();
        });
    });

    it("renders DemoCard component unchanged", () => {
        const { container } = render(<DemoCard demoData={DemoContent[0].demoData} />);
        expect(container).toMatchSnapshot();
    });
});