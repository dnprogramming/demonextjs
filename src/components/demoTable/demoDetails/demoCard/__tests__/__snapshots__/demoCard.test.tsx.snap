// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`DemoCard Component renders DemoCard component unchanged 1`] = `
<div>
  <div
    class="demoGrid"
  >
    <div
      class="demoCard"
    >
      <h2>
        First Card
      </h2>
      <p>
        This is the description for the first card.
      </p>
    </div>
    <div
      class="demoCard"
    >
      <h2>
        Second Card
        <span>
          *
        </span>
      </h2>
      <p>
        Each card is built to be displayed within the demo table system and is customizable.
      </p>
    </div>
    <div
      class="demoCard"
    >
      <h2>
        Third Card
      </h2>
      <p>
        Cards can be active or inactive, and can be marked as new to highlight recent additions.
      </p>
    </div>
  </div>
  <div
    class="demoCardNotation"
  >
    <p>
      <span>
        *
      </span>
       denotes New Demo Description!
    </p>
  </div>
</div>
`;
